server/src/main/java/org/spine3/server/storage/AggregateStorage.java

```diff
@@ -83,52 +86,87 @@ public AggregateEvents read(I aggregateId) {
         return builder.build();
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @throws IllegalArgumentException if events list is empty
+     */
     @Override
-    public void write(I id, AggregateEvents record) {
+    public void write(I id, AggregateEvents events) {
         checkNotClosed();
-
-        final List<EventRecord> list = record.getEventRecordList();
-        for (final EventRecord eventRecord : list) {
-            write(eventRecord);
+        checkNotNull(id, "ID");
+        checkNotNull(events, "events");
+        final List<EventRecord> eventList = events.getEventRecordList();
+        checkArgument(!eventList.isEmpty(), "Event list must not be empty.");
+
+        for (final EventRecord event : eventList) {
+            checkTimestamp(event);
+            final AggregateStorageRecord storageRecord = toStorageRecord(event);
+            writeInternal(id, storageRecord);
         }
     }
 
+    /**
+     * Writes an event record to the storage by an aggregate ID.
+     *
+     * @param id the aggregate ID
+     * @param eventRecord the record to write
+     * @throws NullPointerException if the ID or record is {@code null}
+     * @throws IllegalStateException if the storage is closed
+     */
+    public void writeEvent(I id, EventRecord eventRecord) {
```

Comment:
Please rename the `eventRecord` parameter to `event`. New PR, which contains related renaming is coming.